### DISCUSSION TITLE
Clean cache

### DISCUSSION THREAD

**Discussion by GabRog** [denoland/deno#9419] [Q&A]
Is there any elegant way to clean cache before running program?
Or to enforce deno to ensure that remote imports it is using in exec file, getting from its cache, are the most actual ones? If any of remote imports was updated, is it possible for deno to update the cache?

**Comment by lucacasonato** [comment]
We have a few flags to control this:
```
    -r, --reload=<CACHE_BLOCKLIST>    Reload source code cache (recompile TypeScript)

        --no-remote                   Do not resolve remote modules

        --cached-only                 Require that remote dependencies are already cached
```

You can also find the cache location with `deno info` (if you want to delete the entire cache directory):

```
~ ❯❯❯ deno info
DENO_DIR location: "/Users/lucacasonato/Library/Caches/deno"
Remote modules cache: "/Users/lucacasonato/Library/Caches/deno/deps"
TypeScript compiler cache: "/Users/lucacasonato/Library/Caches/deno/gen"
```
